core/src/main/java/com/cloud/storage/template/QCOW2Processor.java

```diff
@@ -40,6 +40,11 @@
 
     private StorageLayer _storage;
 
+   @Override
+   public FormatInfo process(String templatePath, ImageFormat format, String templateName, long processTimeout) throws InternalErrorException  {
+     return process(templatePath, format, templateName);
```

Comment:
this is not right. it closes the door for implementing this for KVM (and other below) If this is really not applicable for any format but the VMware/OVA bit it should not be on the generic interface.